### PR TITLE
Update rspec to 3.0

### PR DESCRIPTION
**Pull request by gschorkopf**
- Updates gemfile to use 3.0
- Updates rspec syntax
- Updates stubbing syntax

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,11 +1,15 @@
 source 'https://rubygems.org'
 
-gem 'dish' # converts hash to objects with dot syntax
-gem 'rest-client' # http
+# Converts hash to objects with dot syntax
+gem 'dish'
+
+# HTTP requests
+gem 'rest-client'
```
Comment (tdouce) [inline Gemfile]: Adding comments describing the purpose of each gem is a great idea! `<nitpick>`However, these descriptions are not related to updating rspec`</nipick>`. Could they be part of a separate PR?

**Comment by tdouce** [comment]
@gschorkopf Related the gem description comment. There is unfortunately a lot of trailing white space scattered throughout the gem :/ For some reason, I mucked up my my config that highlights them.  Can you remove the trailing white space fixes from this PR so it only includes changes related to updating rspec? I will buy you :cake: if you make a PR specific to removing all the trailing white :)

**Comment by gschorkopf** [comment]
Alrighty.

**Comment by tdouce** [comment]
@gschorkopf Kinda funny. Somebody submitted a PR to Jonathan Wallace's gem this past week for upgrading rspec.
